move golangci-lint to github actions

thaJeztah opened the pull request:
Test run can be seen on https://github.com/thaJeztah/docker/pull/3

thaJeztah committed:
move golangci-lint to github actions

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
gosec: G306: Expect WriteFile permissions to be 0600 or less

    pkg/pidfile/pidfile.go:43:12: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err := ioutil.WriteFile(path, []byte(fmt.Sprintf("%d", os.Getpid())), 0644); err != nil {
                  ^
    pkg/plugins/pluginrpc-gen/main.go:82:33: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        errorOut("error writing file", ioutil.WriteFile(*outputFile, src, 0644))
                                       ^
    testutil/fakecontext/context.go:99:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fp, content, 0644)
               ^
    daemon/graphdriver/graphtest/graphtest_unix.go:310:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(path, data, 0700)
               ^
    layer/filestore.go:243:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fms.getMountFilename(mount, "mount-id"), []byte(mountID), 0644)
               ^
    layer/filestore.go:250:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fms.getMountFilename(mount, "init-id"), []byte(init), 0644)
               ^

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

tianon commented on .github/workflows/validate.yaml:
```diff
@@ -0,0 +1,72 @@
+      - name: install dependencies
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt -q update
+          sudo apt -q install \
```
These are probably printing `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`, so I'd suggest using `apt-get` here instead:

```suggestion
          sudo apt-get -q update
          sudo apt-get -q install \
```

thaJeztah replied:
Damn! Looks like I was caught red-handed doing a copy/pasta from (don't recall where) 😂 I'll update

tianon reviewed (commented):
Nice! I like it :smile:

thaJeztah committed:
golang-ci: update to 1.40.1 and update to use new config options

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
golangci.yml: replace golint with revive as it is deprecated

    WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

tonistiigi commented:
I think the linter should be runnable(same way as it does in CI) without uploading a PR to github. An obvious choice for us would be to run it inside a container/Dockerfile. No objects to invoking that through github actions workflow then.

neersighted commented:
Closed as this is already done with the new actions CI on master/22.06
